lib/replicache.dart

```diff
@@ -291,6 +297,10 @@ class Replicache implements ReadTransaction {
 
     if (reauth && getDataLayerAuth != null) {
       _dataLayerAuth = await getDataLayerAuth();
+      if (_dataLayerAuth != null) {
+        // Try again now instead of waiting for another 5 seconds.
+        return await _beginSync();
```

Comment:
Hm, but now it's going to iloop if the auth fails for some reason. It should really just try once.

Reply:
It will call `getDataLayerAuth` again until it returns null or succeeds. I don't think the current API allows anything better? If you only call it once there is no clear way how to get back to a working state (except to close the Replicache instance and create a new one).

I think we need to rethink how this API works.

Reply:
I mean once per sync invocation. So basically, everytime beginSync is called, if we get back auth required, we try and reauth once and restart beginSync. If that fails, then we let beginSync fail.

Typically I think people will reauth by talking to their own server. I think it is possible for a the batch or client view endpoint to detect that reauth is required, but then the server the user code needs to talk to to actually reauth  not be available.

You're right that we don't have good evidence yet for how this should work so if this suggestion is difficult feel free to just rip this out for now.

Reply:
I'll file an issue and we can revisit.